rush-plugins/rush-http-build-cache-plugin/src/HttpBuildCacheProvider.ts

```diff
@@ -227,13 +239,32 @@ export class HttpBuildCacheProvider implements ICloudBuildCacheProvider {
     });
 
     if (!response.ok) {
-      if (typeof credentials !== 'string' && safeCredentialOptions === CredentialsOptions.Optional) {
+      const isFatal = response.status >= 500 || response.status < 600;
+
+      if (
+        !isFatal &&
+        typeof credentials !== 'string' &&
+        safeCredentialOptions === CredentialsOptions.Optional
+      ) {
         // We tried fetching the resource without credentials and that did not work out
         // Try again but require credentials this time
         // This will trigger the provider to request credentials
+        // This counts as part of the "first attempt", so it is not included in the max attempts
         return await this._http({ ...options, credentialOptions: CredentialsOptions.Required });
       }
 
+      if (options.maxAttempts > 1) {
+        // Pause a bit before retrying in case the server is busy
+        // Add some random jitter to the retry so we can spread out load on the remote service
+        // A proper solution might add exponential back off in case the retry count is high (10 or more)
+        const factor = 1.0 + Math.random(); // A random number between 1.0 and 2.0
+        const retryDelay = Math.floor(factor * this._minHttpRetryDelay);
+
+        await new Promise((resolve) => setTimeout(resolve, retryDelay));
```

Comment:
There exists the node `timers/promises` API that does this more concisely, but we probably need to upgrade `@types/node` repository-wide first.

Reply:
There's also `Async.sleep` from `node-core-library`.

Reply:
:+1: For now moved to `Async.sleep`